core/src/main/java/org/infinispan/security/impl/AuthorizationHelper.java

```diff
@@ -35,27 +38,31 @@
    private final AuditLogger audit;
    private final AuditContext context;
    private final String name;
-   private final ClusterRegistry<String, Subject, Integer> maskCache;
-   private final String maskCacheScope;
+   private final ConcurrentMap<CachePrincipalPair, SubjectACL> aclCache;
 
-   public AuthorizationHelper(GlobalSecurityConfiguration globalConfiguration, AuditContext context, String name, ClusterRegistry<String, Subject, Integer> clusterRegistry) {
+   public AuthorizationHelper(GlobalSecurityConfiguration globalConfiguration, AuditContext context, String name,
+         ConcurrentMap<CachePrincipalPair, SubjectACL> aclCache) {
       this.globalConfiguration = globalConfiguration;
       this.audit = globalConfiguration.authorization().auditLogger();
       this.context = context;
       this.name = name;
-      this.maskCache = clusterRegistry;
-      this.maskCacheScope = AuthorizationManager.class.getSimpleName() + "_" + name;
+      this.aclCache = aclCache;
    }
 
-   public AuthorizationHelper(GlobalSecurityConfiguration globalConfiguration, AuditContext context, String name) {
-      this(globalConfiguration, context, name, null);
+   public AuthorizationHelper(GlobalSecurityConfiguration security, AuditContext context, String name) {
+      this(security, context, name, CollectionFactory.makeBoundedConcurrentMap(10));
    }
 
    public void checkPermission(AuthorizationPermission perm) {
-      checkPermission(null, perm);
+      checkPermission(null, perm, Optional.empty());
```

Comment:
I don't really like mixing `Optional` with old-style nullable parameters, so I'd rather use `null` for both. (Or `Optional` for both, of course.)